name.abuchen.portfolio/src/name/abuchen/portfolio/online/impl/MyDividends24Uploader.java

```diff
@@ -50,21 +53,37 @@ public List<Pair<Integer, String>> getPortfolios() throws IOException
     }
 
     @SuppressWarnings("unchecked")
-    public void upload(Client client, CurrencyConverter converter, String portfolioId) throws IOException
+    public void upload(Client client, CurrencyConverter converter, String portfolioId, Portfolio ppPortfolio)
+                    throws IOException
     {
-        ClientSnapshot snapshot = ClientSnapshot.create(client, converter, LocalDate.now());
-        PortfolioSnapshot portfolio = snapshot.getJointPortfolio();
+
+        Stream<PortfolioTransaction> stream;
+
+        if (ppPortfolio != null)
+        {
+            // Case: one portfolio of PP is selected
+            stream = ppPortfolio.getTransactions().stream();
+        }
+        else
+        {
+            // Case: all portfolios of PP is selected
+            ClientSnapshot snapshot = ClientSnapshot.create(client, converter, LocalDate.now());
+            PortfolioSnapshot portfolio = snapshot.getJointPortfolio();
+            stream = portfolio.getPortfolio().getTransactions().stream();
+
+        }
 
         // Filters out any transactions that do not have an ISIN.
-        List<JSONObject> resultTransactions = portfolio.getPortfolio().getTransactions().stream()
-                        .filter(item -> item.getSecurity().getIsin() != null).map(item -> {
+        List<JSONObject> resultTransactions = stream.filter(item -> item.getSecurity().getIsin() != null)
+                        .map(item -> {
                             double quantity = item.getShares() / Values.Share.divider();
                             double buyingprice = item.getGrossValueAmount() / Values.Amount.divider() / quantity;
                             String purchasedate = item.getDateTime().toString();
                             String isin = item.getSecurity().getIsin();
                             String type = item.getType().isPurchase() ? "buy" : "sell"; //$NON-NLS-1$ //$NON-NLS-2$
 
-                            // Creates a JSONObject for each transaction with
+                            // Creates a JSONObject for each transaction
+                            // with
                             // the calculated data.
                             JSONObject jsonObject = new JSONObject();
```

Comment:
I am not sure if you are aware, but

* because you are reading the raw transactions, the transactions can be in any currency. A client can contain multiple accounts in different currencies, and instruments can be in different currencies. Don't you need a currency symbol in the json object then?
* you are also including the transfer transactions (transfer of instruments between accounts). You would loose this information because it is converted to "buy" and "sell" (which then has impact on FIFO calculations). I would think you would want to skip the transfers (if they are between portfolios that are included in the upload).